File: Course/CS50-SQL-08-Scaling/ProblemSet6/p02-Don’t_Panic!(_Sentimental)_with_Python/dont-panic-python/hack.py
import hashlib
from cs50 import SQL

# print("Hacked!")

db = SQL("sqlite:///dont-panic.db")

str2hash = "hacked!"

result = hashlib.sha512(str2hash.encode()).hexdigest()
row = db.execute(
    """SELECT id, username, password
FROM users
WHERE username='admin';"""
)
db.execute(
    """
           UPDATE "users"
           SET "password" = :password
           WHERE username='admin'
           """,
    password=result,
)
row = db.execute(
    """SELECT id, username, password
FROM users
WHERE username='admin';"""
)

File: Course/CS50-SQL-08-Scaling/ProblemSet6/p03-Don’t_Panic!_(Sentimental)_with_Java/dont-panic-java/Hack.java
import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.util.Base64;

public class Hack {
    public static void main(String[] args) throws Exception {

        String query = """
                UPDATE "users"
                SET "password"= ?
                WHERE "username" = 'admin'
                    ;
                    """;
        var text = "Hacked!";
        MessageDigest digest = MessageDigest.getInstance("SHA-256");
        byte[] hash = digest.digest(text.getBytes(StandardCharsets.UTF_8));
        String encoded = Base64.getEncoder().encodeToString(hash);

        try {
            Connection sqliteConnection = DriverManager.getConnection("jdbc:sqlite:dont-panic.db");
            PreparedStatement sqliteStatement = sqliteConnection.prepareStatement(query);
            sqliteStatement.setString(1, encoded);

            sqliteStatement.executeUpdate();
            sqliteConnection.close();

        } catch (Exception e) {
            // if the error message is "out of memory",
            // it probably means no database file is found
            System.err.println(e.getMessage());
        }

    }
}
